### PR TITLE
feat: Nearest places on the map (BE)

### PR DESCRIPTION
**Pull request by AverchenkoAA**

### CODE REVIEW

**Comment by AverchenkoAA** [comment]
[(feat) Nearest places on the map (BE)](https://trello.com/c/BU1y2uVd/17-feat-nearest-places-on-the-map-be)

File: apps/api/src/app/controllers/maps.controller.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @Query() longit: number
+  ): Promise<Sight> {
+    //point relative to which the nearest points are searched
+    const startPoint: Coordinates = {
+      latitude: latit,
+      longitude: longit
+    };
+
+    //find only one the nearest point from array of points
+    const nearestPoint = await this.geoService.getSightFromPoints(startPoint);
```
Comment (BaturevichNikita) [inline apps/api/src/app/controllers/maps.controller.ts]: You get point, but use get`Sight`FromPoints.
And why from points, if you use one point.

File: apps/api/src/app/controllers/maps.controller.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @Query() longit: number
+  ): Promise<Sight[]> {
+    //point relative to which the nearest points are searched
+    const startPoint: Coordinates = {
+      latitude: latit,
+      longitude: longit
+    };
+
+    //find only one the nearest point from array of points
+    const nearestPoints = await this.geoService.getSightsFromPoints(
```
Comment (BaturevichNikita) [inline apps/api/src/app/controllers/maps.controller.ts]: see above

File: apps/api/src/app/controllers/sights.controller.ts [hunk 3]
```diff
@@ -14,23 +14,26 @@ export class SightsController extends Controller {
...
   @Delete('{id}')
   public async deleteSights(id: string): Promise<string> {
-    let isDeleted = await this.sightsService.deleteSightByID(id);
-    return `Success delete ${id}`;
+    let message = '';
+    const deleteResult: boolean = await this.sightsService.deleteSightByID(id);
+    deleteResult
+      ? (message = `Success delete ${id}`)
+      : (message = `Something went wrong with delete ${id}`);
+    return message;
```
Comment (BaturevichNikita) [inline apps/api/src/app/controllers/sights.controller.ts]: return deleteResult ? `Success delete ${id}` : `Something went wrong with delete ${id}`

File: apps/api/src/app/services/authorization/authorizations.service.ts [hunk 4]
```diff
@@ -6,14 +6,13 @@ import { FirebaseService } from '@mogilev-guide/api/services/firebase';
 export class AuthService {
   @Inject() private firebaseService!: FirebaseService;
 
-  public async getUsersByID(id: string): Promise<User[]> {
+  public async getUsersByID(id: string): Promise<User> {
```
Comment (BaturevichNikita) [inline apps/api/src/app/services/authorization/authorizations.service.ts]: getUsersByID => getUserByID

File: apps/api/src/app/services/sights/sights.service.ts [hunk 5]
```diff
@@ -17,44 +16,50 @@ export class SightsService {
...
-    console.log("");
-    const snapshot = await this.firebaseService.firestore
-      .collection(this.collectionName)
-      .where('id', '==', id)
-      .get();
-    if (!snapshot.docs[0]){
-      return Promise.resolve(snapshot[0]);
-    }
-    return Promise.resolve(snapshot.docs[0].data() as Sight);
+    return await this.getSigthByField('id', id);
```
Comment (BaturevichNikita) [inline apps/api/src/app/services/sights/sights.service.ts]: await is not necessary

File: apps/api/src/app/services/sights/sights.service.ts [hunk 6]
```diff
@@ -17,44 +16,50 @@ export class SightsService {
...
-    }
-    return Promise.resolve(true);
+    return !!snapshot;
+  }
+
+  public async getSightByCoordinates(coordinate: Coordinates): Promise<Sight> {
+    return await this.getSigthByField('coordinates', coordinate);
+  }
+
+  private async getSigthByField(
```
Comment (BaturevichNikita) [inline apps/api/src/app/services/sights/sights.service.ts]: Typo getSigthByField

File: apps/api/src/app/authorization/auth.middleware.ts [hunk 7]
```diff
@@ -54,16 +54,16 @@ export class AuthorizationMiddleware {
...
       },
-      this.getUserID(this.authService)
+      await this.getUserID(this.authService)
     );
   }
 
-  private getUserID(authServ: AuthService) {
-    return async (error: Error, response: express.Response, body: any) => {
-      if (!error && response.statusCode == 200) {
+  private async getUserID(authServ: AuthService) {
```
Comment (BaturevichNikita) [inline apps/api/src/app/authorization/auth.middleware.ts]: Why async? I think it's not necessary.

**Pull request merged**